src/main/java/com/faforever/api/map/MapService.java

```diff
@@ -0,0 +1,348 @@
+    String newNameFolder = "/maps/" + mapData.getNewFolderName();
+    try (Stream<Path> mapFileStream = Files.list(mapData.getNewMapFolder())) {
+      mapFileStream
+          .filter(path -> path.toString().toLowerCase().endsWith(".lua"))
+          .forEach(path -> noCatch(() -> {
+            List<String> collect = Files.readAllLines(path, MAP_CHARSET).stream()
+                .map(line -> line.replaceAll("(?i)" + oldNameFolder, newNameFolder))
+                .collect(Collectors.toList());
+            Files.write(path, collect, MAP_CHARSET);
+          }));
+    }
+  }
+
+
+  private void generatePreview(MapUploadData mapData) throws IOException {
+    String previewFilename = mapData.getNewFolderName() + ".png";
+    generateImage(Paths.get(
+        fafApiProperties.getMap().getMapPreviewPathSmall(), previewFilename),
+        mapData.getNewMapFolder(),
+        fafApiProperties.getMap().getPreviewSizeSmall());
+
+    generateImage(Paths.get(
+        fafApiProperties.getMap().getMapPreviewPathLarge(), previewFilename),
+        mapData.getNewMapFolder(),
+        fafApiProperties.getMap().getPreviewSizeLarge());
+  }
+
+  @SneakyThrows
+  private void zipMapData(MapUploadData progressData) {
+    cleanupBaseDir(progressData);
+    try (ZipOutputStream zipOutputStream = new ZipOutputStream(new BufferedOutputStream(
+        Files.newOutputStream(progressData.getFinalZipFile())))) {
+      Zipper.contentOf(progressData.getBaseDir()).to(zipOutputStream).zip();
+    }
+  }
+
+  private void cleanupBaseDir(MapUploadData progressData) throws IOException {
+    Files.delete(progressData.getUploadedFile());
+    try (Stream<Path> stream = Files.list(progressData.getBaseDir())) {
+      if (stream.count() != 1) {
+        throw new IllegalStateException("Folder containing unknown data: " + progressData.getBaseDir());
+      }
+    }
+  }
+
+  @SneakyThrows
+  private void generateImage(Path target, Path baseDir, int size) {
+    Image image = PreviewGenerator.generatePreview(baseDir, size, size);
+    JavaFxUtil.writeImage(image, target, "png");
+  }
+
+
+  private boolean cleanup(MapUploadData mapData) {
+    return FileSystemUtils.deleteRecursively(mapData.getBaseDir().toFile());
+  }
+
+  @Data
+  private class MapUploadData {
+    private String uploadFileName;
+    private String uploadFolderName;
+    private String newFolderName;
+    private Path uploadedFile;
+    private Path baseDir;
+    private Path originalMapFolder;
+    private Path newMapFolder;
+    private Path finalZipFile;
+    private LuaValue luaRoot;
+    private Map mapEntity;
+    private MapVersion mapVersionEntity;
+    private Player authorEntity;
+    private boolean isRanked;
+
+    public LuaValue getLuaScenarioInfo() {
+      if (getLuaRoot() == null) {
+        throw new IllegalStateException("*_scenario.lua parse result not available");
+      }
+      return getLuaRoot().get("ScenarioInfo");
```

Comment:
Field access could be used instead of repeated getter call. Also, pay attention to IntelliJ warnings.